client/pp-display/src/pp-box.tsx

```diff
@@ -16,8 +28,8 @@ const ADDED_DEPTH_FACTOR = 10;
 
 const PpBox: FunctionComponent<PpBoxProps> = (props) => {
     
-    const {mode, depth, coqCss, id, indent, breaks, boxChildren, hovered, maxDepth, addedDepth} = props;
-    const [hide, setHide] = useState<boolean>(depth >= maxDepth);
+    const {mode, depth, coqCss, id, indent, breaks, boxChildren, parentHide, hovered, maxDepth, addedDepth} = props;
+    const [selfHide, setSelfHide] = useState<HideStates>(ComputeHideState(depth >= maxDepth ? HideStates.HIDE : HideStates.UNHIDE, parentHide));
     const [depthOpen, setDepthOpen] = useState<number>(addedDepth);
```

Comment:
One thing I might add, I was a bit confused about the `depthOpen` variables usage, so maybe there is a simpler solution to the "expand all the way" problem using that variable?

Reply:
```depthOpen``` controls at what depth an ellipsis will open. So when you click on an ellipsis, instead of only unfolding it's child, it will go down ```depthOpen``` which is more comfortable than having to unfold one by one.

Reply:
Sorry to clarity, ```depthOpen``` tracks the current depth at which an ellipsis happens, so the depth until which eveything is open. The ```ADDED_DEPTH_FACTOR``` just allows us to increment this in steps larger than 1.